stage0/src/main.rs

```diff
@@ -17,15 +17,139 @@
 #![no_std]
 #![no_main]
 
-use core::panic::PanicInfo;
-use x86_64::instructions::{hlt, interrupts::int3};
-
+use core::{arch::asm, ffi::c_void, panic::PanicInfo};
+use x86_64::{
+    instructions::{hlt, interrupts::int3, segmentation::Segment},
+    registers::{
+        control::{Cr3, Cr3Flags},
+        segmentation::*,
+    },
+    structures::{
+        gdt::{Descriptor, GlobalDescriptorTable, SegmentSelector},
+        idt::InterruptDescriptorTable,
+        paging::{
+            page_table::{PageTable, PageTableFlags},
+            PhysFrame,
+        },
+    },
+    PhysAddr, VirtAddr,
+};
 mod asm;
 
+// Magic constants where various data structures will be stored.
+// Many of these are set to be same as with crosvm; see
+// https://google.github.io/crosvm/appendix/memory_layout.html for more details.
+const BOOT_GDT_ADDR: PhysAddr = PhysAddr::new(0x1500);
+// The Rust GlobalDescriptorTable has 8 entries; the one provided by crosvm has 4.
```

Comment:
nit: I assume this comment means that the addres of the IDT is at a different place compared to crosvm, but not 100% sure. Perhaps adding a bit of additional clarification to the comment will make the intention clearer.

Reply:
Yes, crosvm places the IDT at 0x1520. I've expanded the comment.